indexing-service/src/test/java/io/druid/indexing/overlord/http/security/SecurityResourceFilterTest.java

```diff
@@ -0,0 +1,115 @@
+package io.druid.indexing.overlord.http.security;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.sun.jersey.spi.container.ResourceFilter;
+import io.druid.indexing.common.task.NoopTask;
+import io.druid.indexing.common.task.Task;
+import io.druid.indexing.overlord.TaskStorageQueryAdapter;
+import io.druid.indexing.overlord.http.OverlordResource;
+import io.druid.indexing.worker.http.WorkerResource;
+import io.druid.server.http.security.AbstractResourceFilter;
+import io.druid.server.http.security.ResourceFilterTestHelper;
+import org.easymock.EasyMock;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import java.lang.reflect.Field;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class SecurityResourceFilterTest extends ResourceFilterTestHelper
+{
+  @Parameterized.Parameters
+  public static Collection<Object[]> data()
+  {
+    return ImmutableList.copyOf(
+        Iterables.concat(
+            getRequestPaths(OverlordResource.class),
+            getRequestPaths(WorkerResource.class)
+        )
+    );
+  }
+
+  private final String requestPath;
+  private final String requestMethod;
+  private final ResourceFilter resourceFilter;
+  private final TaskStorageQueryAdapter tsqa;
+  private final Task noopTask = new NoopTask(null, 0, 0, null, null, null);
+
+  public SecurityResourceFilterTest(
+      String requestPath,
+      String requestMethod,
+      ResourceFilter resourceFilter
+  )
+  {
+    this.requestPath = requestPath;
+    this.requestMethod = requestMethod;
+    this.resourceFilter = resourceFilter;
+    tsqa = EasyMock.createStrictMock(TaskStorageQueryAdapter.class);
+  }
+
+  @Before
+  public void setUp() throws Exception
+  {
+    if (resourceFilter instanceof TaskResourceFilter) {
+      EasyMock.expect(tsqa.getTask(EasyMock.anyString())).andReturn(Optional.of(noopTask)).anyTimes();
+      Field tsqaField = TaskResourceFilter.class.getDeclaredField("taskStorageQueryAdapter");
```

Comment:
The reason I did not create a parametrized constructor is because this class is instantiated by jersey so I was not sure if it will work or even if it works now, will it work in future versions of jersey ?